rqd/rqd/rqcore.py

```diff
@@ -421,12 +421,14 @@ def run(self):
                                                  runFrame.frame_name)
             runFrame.log_dir_file = os.path.join(runFrame.log_dir, runFrame.log_file)
 
-            try: # Exception block for all exceptions
-                # Do everything as launching user
-                runFrame.gid = rqd.rqconstants.LAUNCH_FRAME_USER_GID
+            try:  # Exception block for all exceptions
+
+                # Change to frame user if needed:
+                if runFrame.HasField("uid"):
```

Comment:
We should probably also have a check here to ensure that if RQD is running as root then a `uid` must be provided. This would continue the current behaviour of disallowing running frames as root.

Reply:
I don't really strongly about enforcing such a requirement. Certainly switching to root by passing a UID should not be possible, but if a user wants to run RQD as root and runs jobs as root, I think that's their choice.

What happens in a Docker environment, i.e. if someone has built a container where that user appears as root to RQD?